src/Neo.VM/Types/CompoundType.cs

```diff
@@ -65,7 +66,12 @@ public sealed override bool GetBoolean()
         /// <exception cref="NotSupportedException">This method always throws the exception.</exception>
         public override int GetHashCode()
         {
-            throw new NotSupportedException();
+            var h = new HashCode();
+            h.Add(Count);
+            h.Add(Type);
+            foreach (var item in SubItems)
```

Comment:
If there is a circular reference here, it will cause problems.

Reply:
fixed